bpf/lib/encap.h

```diff
@@ -32,6 +34,9 @@ __encap_with_nodeid(struct __ctx_buff *ctx, __u32 src_ip, __be16 src_port,
 	if (seclabel == HOST_ID)
 		seclabel = LOCAL_NODE_ID;
 
+	if (seclabel > ENCAP_MAX_VNI)
+		return DROP_INVALID_IDENTITY;
+
```

Comment:
To clarify - is this patch meant to address the CIDR-identity concern from https://github.com/cilium/cilium/pull/29923#discussion_r1429574175, or are you planning to add more graceful handling?

Reply:
for now just to make sure we don't pass down local identities. This is the only place where we return the invalid ID drop reason, so should be already a good way to detect if this happens

Reply:
But you could easily catch that at the specific location where `Cluster-aware-addressing` sets the forwarded identity? And thus self-document that it's not supported yet (and why).